internal/observability/events/eventer_retry.go

```diff
@@ -0,0 +1,79 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/eventlogger"
+	"github.com/hashicorp/go-multierror"
+)
+
+const (
+	// stdRetryCount is the standard number of times for retry when sending events
+	stdRetryCount = 3
+)
+
+type backoff interface {
+	duration(attemptNumber uint) time.Duration
+}
+
+type expBackoff struct{}
+
+func (b expBackoff) duration(attempt uint) time.Duration {
+	r := rand.Float64()
+	return time.Millisecond * time.Duration(math.Exp2(float64(attempt))*5*(r+0.5))
+}
+
+type retryInfo struct {
+	retries int
+	backoff time.Duration
+}
+
+type sendHandler func() (eventlogger.Status, error)
+
+func (e *Eventer) retrySend(ctx context.Context, retries uint, backOff backoff, handler sendHandler) error {
```

Comment:
I feel like this would do well with some comments

Reply:
Any specific comments?  How about:  `Hi Mom!  I hope things are well and I miss you.  Today, Finn and I took a nap after lunch.  We really should do that every day.  BTW, Sarah was working on her day off.  She'd be giving me so much grief if I did that.  Too funny!  Well, gotta go.  Love you Mom!`

Reply:
I added a more appropriate comment.  ty!